src/main/java/frc/robot/subsystems/VisionSubsystem.java

```diff
@@ -220,21 +221,35 @@ private void process(
       }
       // makes a field pose for logging
       var FieldPose = FieldPose3d.toPose2d();
+      int num_targets = result.targets.size();
       // gets distance
       var Distance =
           PhotonUtils.getDistanceToPose(
               FieldPose,
               // gets closed tag and gets distance
               fieldLayout.getTagPose(result.getBestTarget().getFiducialId()).get().toPose2d());
       // makes a pose that vision sees
+      double all_of_the_distances = 0;
+      for (PhotonTrackedTarget target : result.targets) {
+        var dist =
+            PhotonUtils.getDistanceToPose(
+                FieldPose,
+                // gets closed tag and gets distance
+                fieldLayout.getTagPose(target.getFiducialId()).get().toPose2d());
+        all_of_the_distances += dist;
+      }
+      all_of_the_distances = all_of_the_distances / num_targets;
+      double stdDevFactor = Math.pow(all_of_the_distances, 2.0) / num_targets;
+      double linearStdDev = 0.02 * stdDevFactor;
+      double angularStdDev = 0.06 * stdDevFactor;
       aprilTagsHelper.addVisionMeasurement(
-          // field pose
           FieldPose,
-          // timestamp
-          TimestampSeconds,
-          // start with STANDARD_DEVS, and for every meter of distance past 1 meter, add another
-          // DISTANCE_SC_STANDARD_DEVS to the standard devs
-          DISTANCE_SC_STANDARD_DEVS.times(Math.max(0, Distance - 1)).plus(STANDARD_DEVS));
+          RawTimestampSeconds,
+
+          //// Use one of these, first one is current second is what advantage kit example is
```

Comment:
Please do the math (Excel is good at this) to compare the two methods for various tag distances & see how that affects the values in the vector below. I'd expect this to also depend on the exact camera & lens plus the pipeline configuration (mostly resolution & shutter speed) since I believe those are the primary vision contributors to noise in the data. Robot movement (rotation especially) would also significantly affect noise, so it'd be interesting to see if using that to scale up the stdDev values helps (by decreasing confidence in position values read while the camera's field of view is rapidly changing).